cmd/finch/nerdctl_remote.go

```diff
@@ -501,3 +539,134 @@ func handleEnvFile(fs afero.Fs, systemDeps NerdctlCommandSystemDeps, arg, arg2 s
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" || runtime.GOOS != "darwin" {
+		logrus.Warn("handleCache returning unchanged: ", arg)
+		logrus.Warn("handleCache info: ", fc.Mode, *fc.Mode, runtime.GOOS)
+		return arg
+	}
+
+	if strings.Contains(arg, "consistency") {
+		logrus.Warn("handleCache manipulating: ", arg)
+		arg = strings.Replace(arg, ",consistency=cached", "", 1)
+		arg = strings.Replace(arg, ",consistency=delegated", "", 1)
+		arg = strings.Replace(arg, ",consistency=consistent", "", 1)
+		logrus.Warn("handleCache returning: ", arg)
+	}
+	return arg
+}
+
+func handleLoad(fc *config.Finch, args []string) []string {
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" || args == nil {
+		return args
+	}
+
+	if *fc.Mode == "dockercompat" {
+		logrus.Warn("appending --output-type!!")
+		logrus.Warn("args before appending", args)
+		args = append(args, "--output=type=docker")
+		logrus.Warn("args after appending", args)
+	}
+
+	return args
+}
+
+func handleDockerCompatInspect(fc *config.Finch, args []string) []string {
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" {
+		return args
+	}
+
+	newArgList := []string{}
+	isInspect := false
+	inspectIndex := -1
+	skipNext := false
+	inspectType := ""
+
+	for idx, arg := range args {
+		if skipNext {
+			skipNext = false
+			continue
+		}
+		if arg == "inspect" {
+			isInspect = true
+			inspectIndex = idx
+			newArgList = append(newArgList, arg)
+		}
+
+		if strings.Contains(arg, "--type") && (idx < len(args)-1) && isInspect {
+			if strings.Contains(arg, "=") {
+				inspectType = strings.Split(arg, "=")[1]
+			} else {
+				inspectType = args[idx+1]
+			}
+			if arg == "--type" {
+				skipNext = true
+			}
+			continue
+		}
+		newArgList = append(newArgList, arg)
+	}
+
+	if !isInspect {
+		return args
+	}
+
+	switch inspectType {
+	case "container":
+		dcTrailingArgs := append([]string{"--mode=dockercompat"}, newArgList[inspectIndex+2:]...)
+		dcMidArgs := append([]string{inspectType}, append([]string{newArgList[inspectIndex+1]}, dcTrailingArgs...)...)
+		newArgList = append(newArgList[:inspectIndex], dcMidArgs...)
+	case "":
+		break
+	default:
+		newArgList = append(newArgList[:inspectIndex], append([]string{inspectType}, newArgList[inspectIndex+1:]...)...)
+	}
+
+	return newArgList
+}
+
+func handleBuildx(fc *config.Finch, limaArgs []string) (bool, []string) {
+	buildx := false
+	skipCmd := true
+	var newLimaArgs []string
+	buildxSubcommands := []string{"bake", "create", "debug", "du", "imagetools", "inspect", "ls", "prune", "rm", "stop", "use", "version"}
+
+	if fc == nil || fc.Mode == nil || *fc.Mode != "dockercompat" {
+		return !skipCmd, limaArgs
+	}
+
+	for _, arg := range limaArgs {
+		if arg == "buildx" {
+			buildx = true
+			newLimaArgs = append(newLimaArgs, "build")
+			logrus.Warn("buildx is not supported. using standard buildkit instead...")
+			continue
+		}
+
+		// Find standard vsc container identifier to truncate to less than 76chars
```

Comment:
this change we should not be doing and atleast not here, any change to identifier is a risky change and can impact other things, 

Tracing the origin:
This portion is from https://github.com/devcontainers/cli/blob/9ba1fdaa11dee087b142d33e4ac13c5788392e34/src/spec-node/utils.ts#L527

and  that is getting called in: https://github.com/devcontainers/cli/blob/9ba1fdaa11dee087b142d33e4ac13c5788392e34/src/spec-node/singleContainer.ts#L121


We can configure this under devcontainer.json

config {
 image: <name>
}
This can be part of our docs instead of doing it like this.

Reply:
You were correct about the devcontainers code reference.
I also followed it to the `getFolderImageName` and found the specific concatenation that was creating the "extra long name" (including folderHash): https://github.com/devcontainers/cli/blob/9ba1fdaa11dee087b142d33e4ac13c5788392e34/src/spec-node/utils.ts#L527